### DISCUSSION TITLE
ExpandedNodeId

### DISCUSSION THREAD

**Discussion by nict17** [FreeOpcUa/opcua-asyncio#738] [Q&A]
Hello,
Im currently working on a NodeSet-Model for an Aggregation-Server and therefore ive been using Siemens OPC UA Modeler (SiOME). To reference the right Node  from the Server which should be aggregated. Ive been  using the `ExpandedNodeId`-DataType on a BaseVariable. 
According to the [Spec](https://reference.opcfoundation.org/v104/Core/docs/Part4/7.11/) for common Parameter Types  the `ExpandedNodeId` extends the normal NodeId by the information of the underlying NamespaceIndex **or** the NamespaceUri as a string, wich is the one ive been looking for.
After importing the Model with the integrated `XMLImporter` i realised, that the created Node was of the right Data-Type, however the attributes incl. values were not there. Im not quite sure if this is a Problem of the Export out of SiOME or of the XMLImporter.

Could you please give me a hint, whether how to implement it correctly in the XML or how to create a fast workaround, if it is an issue of the XMLImporter?

Best regards,
Nico 

This is the export out of SiOME which will be imported into my Python-Implementation:

![image](https://user-images.githubusercontent.com/95033790/143451529-495c9025-b455-4608-aa5e-301a14b6b5e4.png)

**Comment by schroeder-** [comment]
Values of Type XmlElement, ExpandedNodeId, QualifiedName and StatusCode are not supported by the xmlimporter atm.

**Comment by oroulet** [comment]
if that is only the export failling then it should be easy to fix somewhere here: https://github.com/FreeOpcUa/opcua-asyncio/blob/933f4cb032b88d45cde66e8d9bd6163218982bcf/asyncua/common/xmlexporter.py#L182

**Comment by xydan83** [comment]
I was creating a discussion on a similar topic. #673

**Comment by schroeder-** [comment]
I did a quickfix in #739 this needs some tests and i will also add more missing types.
